models/gif_cond_exp_multisynapse.cpp

```diff
@@ -0,0 +1,654 @@
+
+void
+nest::gif_cond_exp_multisynapse::Parameters_::get( DictionaryDatum& d ) const
+{
+  def< double >( d, names::I_e, I_e_ );
+  def< double >( d, names::E_L, E_L_ );
+  def< double >( d, names::g_L, g_L_ );
+  def< double >( d, names::C_m, c_m_ );
+  def< double >( d, names::V_reset, V_reset_ );
+  def< double >( d, names::Delta_V, Delta_V_ );
+  def< double >( d, names::V_T_star, V_T_star_ );
+  def< double >( d, names::lambda_0, lambda_0_ * 1000.0 ); // convert to 1/s
+  def< double >( d, names::t_ref, t_ref_ );
+  def< double >( d, names::E_ex, E_ex_ );
+  def< double >( d, names::E_in, E_in_ );
+  def< int >( d, "n_synapses", num_of_receptors_ );
+  def< bool >( d, names::has_connections, has_connections_ );
+
+  ArrayDatum tau_syn_ad( tau_syn_ );
+  def< ArrayDatum >( d, names::taus_syn, tau_syn_ad );
+
+  ArrayDatum tau_sfa_list_ad( tau_sfa_ );
+  def< ArrayDatum >( d, names::tau_sfa, tau_sfa_list_ad );
+
+  ArrayDatum q_sfa_list_ad( q_sfa_ );
+  def< ArrayDatum >( d, names::q_sfa, q_sfa_list_ad );
+
+  ArrayDatum tau_stc_list_ad( tau_stc_ );
+  def< ArrayDatum >( d, names::tau_stc, tau_stc_list_ad );
+
+  ArrayDatum q_stc_list_ad( q_stc_ );
+  def< ArrayDatum >( d, names::q_stc, q_stc_list_ad );
+}
+
+void
+nest::gif_cond_exp_multisynapse::Parameters_::set( const DictionaryDatum& d )
+{
+  updateValue< double >( d, names::I_e, I_e_ );
+  updateValue< double >( d, names::E_L, E_L_ );
+  updateValue< double >( d, names::g_L, g_L_ );
+  updateValue< double >( d, names::C_m, c_m_ );
+  updateValue< double >( d, names::V_reset, V_reset_ );
+  updateValue< double >( d, names::Delta_V, Delta_V_ );
+  updateValue< double >( d, names::V_T_star, V_T_star_ );
+
+  if ( updateValue< double >( d, names::lambda_0, lambda_0_ ) )
+  {
+    lambda_0_ /= 1000.0; // convert to 1/ms
+  }
+
+  updateValue< double >( d, names::t_ref, t_ref_ );
+  updateValue< double >( d, names::E_ex, E_ex_ );
+  updateValue< double >( d, names::E_in, E_in_ );
+
+  updateValue< std::vector< double > >( d, names::tau_sfa, tau_sfa_ );
+  updateValue< std::vector< double > >( d, names::q_sfa, q_sfa_ );
+  updateValue< std::vector< double > >( d, names::tau_stc, tau_stc_ );
+  updateValue< std::vector< double > >( d, names::q_stc, q_stc_ );
+
+  std::vector< double > tau_tmp;
+  if ( updateValue< std::vector< double > >( d, names::taus_syn, tau_tmp ) )
+  {
+    for ( size_t i = 0; i < tau_tmp.size(); ++i )
+    {
+      if ( tau_tmp.size() < tau_syn_.size() && has_connections_ == true )
+        throw BadProperty(
+          "The neuron has connections, therefore the number of ports cannot be "
+          "reduced." );
+
+      if ( tau_tmp[ i ] <= 0 )
+        throw BadProperty(
+          "All synaptic time constants must be strictly positive" );
+
+      if ( tau_tmp[ i ] == ( c_m_ / g_L_ ) )
+        throw BadProperty(
+          "Membrane and synapse time constant(s) must differ. See note in "
+          "documentation." );
```

Comment:
As Hans mentioned, we do not need to check this condition (propagator checks this condition itself). Therefore, I removed these lines.